processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -127,9 +128,16 @@ public SearchQueryRunner(Segment segment)
         if (bitmapIndex != null) {
           for (int i = 0; i < bitmapIndex.getCardinality(); ++i) {
             String dimVal = Strings.nullToEmpty(extractionFn.apply(bitmapIndex.getValue(i)));
-            if (searchQuerySpec.accept(dimVal) &&
-                bitmapFactory.intersection(Arrays.asList(baseFilter, bitmapIndex.getBitmap(i))).size() > 0) {
-              retVal.add(new SearchHit(dimension.getOutputName(), dimVal));
+            if (!searchQuerySpec.accept(dimVal)) {
+              continue;
+            }
+            ImmutableBitmap bitmap = bitmapIndex.getBitmap(i);
+            if (bitmapFactory.intersection(Arrays.asList(baseFilter, bitmap)).size() > 0) {
+              MutableInt counter = new MutableInt(bitmap.size());
```

Comment:
Should this get the size() of the intersected bitmap, instead of the base bitmap? Seems like size() on the base bitmap would result in a count too high if some rows were eliminated by the filter.